tools/chpl-language-server/test/test_server.py

```diff
@@ -0,0 +1,611 @@
+            assert results is not None
+            if isinstance(results, CompletionList):
+                results = results.items
+            result_names = [r.label for r in results]
+
+            for symbol in global_symbols:
+                assert symbol in result_names
+
+        assert len(client.diagnostics) == 0
+
+
+@pytest.mark.asyncio
+async def test_go_to_definition_simple(client: LanguageClient):
+    """
+    Basic test cases for go-to-definition.
+    """
+
+    file = """
+
+           var x: int = 5;
+           var y: int = x;
+           var z: int = x + y;
+           {
+             var x = 10;
+             var y = x;
+           }
+           """
+
+    with source_file(client, file) as doc:
+        # Definitions link to themselves
+        await check_goto_decl_def(client, doc, pos((0, 4)), pos((0, 4)))
+        await check_goto_decl_def(client, doc, pos((1, 4)), pos((1, 4)))
+        await check_goto_decl_def(client, doc, pos((2, 4)), pos((2, 4)))
+        await check_goto_decl_def(client, doc, pos((4, 6)), pos((4, 6)))
+        await check_goto_decl_def(client, doc, pos((5, 6)), pos((5, 6)))
+
+        # References link to their definitions
+        await check_goto_decl_def(client, doc, pos((1, 13)), pos((0, 4)))
+        await check_goto_decl_def(client, doc, pos((2, 13)), pos((0, 4)))
+        await check_goto_decl_def(client, doc, pos((2, 17)), pos((1, 4)))
+        await check_goto_decl_def(client, doc, pos((5, 10)), pos((4, 6)))
+
+        # Other things don't go anywhere
+        # 'var' tokens
+        await check_goto_decl_def(client, doc, pos((0, 2)), None)
+        await check_goto_decl_def(client, doc, pos((1, 2)), None)
+        await check_goto_decl_def(client, doc, pos((2, 2)), None)
+        await check_goto_decl_def(client, doc, pos((4, 4)), None)
+        await check_goto_decl_def(client, doc, pos((5, 4)), None)
+        # numbers
+        await check_goto_decl_def(client, doc, pos((0, 13)), None)
+        await check_goto_decl_def(client, doc, pos((4, 10)), None)
+
+        assert len(client.diagnostics) == 0
+
+
+@pytest.mark.asyncio
+async def test_go_to_definition_use_standard(client: LanguageClient):
+    """
+    Ensure that go-to-definition works on standard module symbols
+    that are used or imported.
+    """
+
+    file = """
+           use IO;
+           use List, Map;
+           import Time;
+           """
+
+    mod_IO = standard_module("standard/IO.chpl")
+    mod_List = standard_module("standard/List.chpl")
+    mod_Map = standard_module("standard/Map.chpl")
+    mod_Time = standard_module("standard/Time.chpl")
+
+    with source_file(client, file) as doc:
+        await check_goto_decl_def(client, doc, pos((0, 4)), mod_IO)
+        await check_goto_decl_def(client, doc, pos((1, 4)), mod_List)
+        await check_goto_decl_def(client, doc, pos((1, 10)), mod_Map)
+        await check_goto_decl_def(client, doc, pos((2, 8)), mod_Time)
```

Comment:
I think these should be more specific. `goto def` on `IO` should go to `module IO {`